tests.py

```diff
@@ -15,12 +16,23 @@
 
 LOGGING_LEVEL=logging.ERROR
 
-TESTING_IN_FILE_CSV = "testing/testing_file_in.csv"
-TESTING_OUT_FILE_CSV = "testing/testing_file_out.csv"
-TESTING_OUT_FILE_CSV_2 = "testing/testing_file_out2.csv"
+TESTING_IN_FILE_CSV = "testing/in.csv"
```

Comment:
instead of using fixed test files in the working directory, a temprary directory should be created for this. After the tests, all files should be removed (or simply the entire directory)

Reply:
but where should the input for the devices come from? they have to be stored somewhere in the repository?

Reply:
If you want to read from existing files, then I guess having test files in the repository is good enough. An alternative would be to read samples from a python array, output them to a file, and then read from that file back into an array, and compare the two arrays. In any case, any new created file should be in an external, temporary folder.